src/http_client.ts

```diff
@@ -152,13 +152,21 @@ export class HttpClient {
             this.stats.requests++;
             let response: ApifyResponse;
             const requestIsStream = isStream(config.data);
+
             try {
                 if (requestIsStream) {
                     // Handling redirects is not possible without buffering - part of the stream has already been sent and can't be recovered
                     // when server sends the redirect. Therefore we need to override this in Axios config to prevent it from buffering the body.
                     // see also axios/axios#1045
                     config = { ...config, maxRedirects: 0 };
                 }
+
+                // Increase timeout with each attempt. Max timeout is bounded by the client timeout.
+                config.timeout = Math.min(
+                    this.timeoutMillis,
+                    config.timeout ?? this.timeoutMillis * 2 ** (attempt - 1),
```

Comment:
Nullish coalescing has lower precedence than multiplication. 

This would evaluate to `config.timeout ?? (this.timeoutMillis * 2 ** (attempt - 1))` (see the parentheses), which - in case `config.timeout` is `undefined` - would cause the `.min` operator to always resolve to `this.timeoutMillis` (the first expression).

Did you want to use `(config.timeout ?? this.timeoutMillis) * 2 ** (attempt - 1)`? (see parentheses)